src/proof/lfsc/lfsc_node_converter.cpp

```diff
@@ -604,11 +601,43 @@ TypeNode LfscNodeConverter::postConvertType(TypeNode tn)
 
 std::string LfscNodeConverter::getNameForUserName(const std::string& name)
 {
+  std::stringstream ssan;
   std::stringstream ss;
-  ss << "cvc." << name;
+  // We also must sanitize symbols that are not allowed in LFSC identifiers
+  // here. For the sake of generating unique symbols, we record the santization
+  // as a prefix before the "." separating the user name. Thus, e.g.
+  //   cvc.|a b|
+  // is converted to:
+  //   cvc_32@1.ab
```

Comment:
I think that it would be good if the sanitized string started with an `@` or a `.` because the SMT-LIB standard specifies that those are reserved for solver-internal uses. This would hopefully ensure that the sanitized symbol names don't accidentally clash with symbols introduced by the user.

Reply:
So the main purpose of this sanitization is to ensure user symbols do not clash with *lfsc* definitions (e.g. proof rule names).

Previously for user symbols, we just did `cvc.X`, i.e. prefix the symbol with "cvc.".

Now we do `cvc_[c1]@[p1]..._[cn]@[pn].Y` where c1...cn are code points and p1...pn are positions and Y is the result of removing all bad characters from X.

This should be injective.  It is unique from LFSC definitions, which dont have `cvc` as a prefix.

I'm not sure why we'd need to special case symbols that start with `@`?

Reply:
See my reply to the comment below.